docs/docs/guides/03-typescript.md

```diff
@@ -2,4 +2,226 @@
 title: Typescript
 ---
 
-🚧 coming soon 🚧
+# Development Guidelines
+
+This document provides guidelines for developing on the Strapi codebase using TypeScript. It covers various topics, including types location, generics, breaking changes, factories and more.
+
+## 1. Type Location
+
+When working with types on the Strapi codebase, it is essential to consider the location of types. These are the three common approaches and situations we document:
+
+### a. Colocation
+
+In the colocation approach, type definitions are colocated with the corresponding code. This means that types are defined in the same file or module where they are used.
+
+**This approach is suitable when the types are specific to a particular module**.
+
+_Note: In some situation, colocated types can be exported too if needed by other modules_
```

Comment:
```suggestion
_Note: In some situations, colocated types can still be exported if needed by other modules_
```
But also, what are those situations?

Reply:
This for instance https://github.com/strapi/strapi/pull/17104/files#r1242366691

@joshuaellis I think props falls in this category

Reply:
6913093336838e93042f925aa4fb2c82d0a7db1d

Reply:
6913093336838e93042f925aa4fb2c82d0a7db1d